drivers/power/ltc2992/ltc2992.h

```diff
@@ -0,0 +1,450 @@
+#define LTC2992_REG_DSENSE_MAX_THRESH(x)	(LTC2992_REG_DSENSE1_MAX_THRESH + ((x) * 0x32))
+#define LTC2992_REG_DSENSE_MIN_THRESH(x)	(LTC2992_REG_DSENSE1_MIN_THRESH + ((x) * 0x32))
+#define LTC2992_REG_SENSE(x)			(LTC2992_REG_SENSE1 + ((x) * 0x32))
+#define LTC2992_REG_SENSE_MAX(x)		(LTC2992_REG_SENSE1_MAX + ((x) * 0x32))
+#define LTC2992_REG_SENSE_MIN(x)		(LTC2992_REG_SENSE1_MIN + ((x) * 0x32))
+#define LTC2992_REG_SENSE_MAX_THRESH(x)		(LTC2992_REG_SENSE1_MAX_THRESH + ((x) * 0x32))
+#define LTC2992_REG_SENSE_MIN_THRESH(x)		(LTC2992_REG_SENSE1_MIN_THRESH + ((x) * 0x32))
+#define LTC2992_REG_POWER_FAULT(x)		(LTC2992_REG_FAULT1 + ((x) * 0x32))
+#define LTC2992_REG_SENSE_FAULT(x)		(LTC2992_REG_FAULT1 + ((x) * 0x32))
+#define LTC2992_REG_DSENSE_FAULT(x)		(LTC2992_REG_FAULT1 + ((x) * 0x32))
+
+#define LTC2992_OFFSET_CAL_MSK			NO_OS_BIT(7)
+#define LTC2992_MEASURE_MODE_MSK		NO_OS_GENMASK(6,5)
+#define LTC2992_VSEL_CON_MSK			NO_OS_GENMASK(4,3)
+#define LTC2992_VSEL_SNAP_MSK			NO_OS_GENMASK(2,0)
+#define LTC2992_POWER_ALERT_MSK			NO_OS_GENMASK(7,6)
+#define LTC2992_DSENSE_ALERT_MSK		NO_OS_GENMASK(5,4)
+#define LTC2992_SENSE_ALERT_MSK			NO_OS_GENMASK(3,2)
+#define LTC2992_ISUM_ALERT_MSK			NO_OS_GENMASK(3,2)
+#define LTC2992_PSUM_ALERT_MSK			NO_OS_GENMASK(1,0)
+#define LTC2992_POWER_FAULT_MSK			NO_OS_GENMASK(7,6)
+#define LTC2992_DSENSE_FAULT_MSK		NO_OS_GENMASK(5,4)
+#define LTC2992_SENSE_FAULT_MSK			NO_OS_GENMASK(3,2)
+#define LTC2992_ISUM_FAULT_MSK			NO_OS_GENMASK(3,2)
+#define LTC2992_PSUM_FAULT_MSK			NO_OS_GENMASK(1,0)
+#define LTC2992_12B_SENSE_VAL_MSK		NO_OS_GENMASK(15,4)
+#define LTC2992_12B_POWER_VAL_MSK		NO_OS_GENMASK(23,0)
+#define LTC2992_8B_SENSE_VAL_MSK		NO_OS_GENMASK(15,8)
+#define LTC2992_8B_POWER_VAL_MSK		NO_OS_GENMASK(23,8)
+
+#define LTC2992_G1_FAULT_ALERT_MSK		0x03
+#define LTC2992_G2_FAULT_ALERT_MSK		0x03
+#define LTC2992_G3_FAULT_ALERT_MSK		0xC0
+#define LTC2992_G4_FAULT_ALERT_MSK		0x30
+
+#define LTC2992_ALERT_CLR_BIT			NO_OS_BIT(7)
+#define LTC2992_READ_CTRL_BIT			NO_OS_BIT(5)
+#define LTC2992_STUCK_BUS_TIMEOUT_BIT		NO_OS_BIT(4)
+#define LTC2992_PEAK_RST_BIT			NO_OS_BIT(3)
+#define LTC2992_RESET_BIT			NO_OS_BIT(0)
+#define LTC2992_G4_ALERT_PIN_BIT		NO_OS_BIT(7)
+#define LTC2992_G1_CTRL_BIT			NO_OS_BIT(7)
+#define LTC2992_G2_CTRL_BIT			NO_OS_BIT(6)
+#define LTC2992_G3_CTRL_BIT			NO_OS_BIT(0)
+#define LTC2992_G4_CTRL_BIT			NO_OS_BIT(6)
+
+#define LTC2992_MFR_ID_VALUE			0x62
+#define LTC2992_RESOLUTION_POS			7
+
+/******************************************************************************/
+/******************* Driver pre-compile time settings *************************/
+/******************************************************************************/
+
+/**
+ * @brief   Allow access to LTC2992 GPIOs.
+ * @details If set to @p TRUE, driver functions for GPIO can be accessed.
+ * @note    The default is @p TRUE.
+ */
+#ifndef LTC2992_USE_GPIO
+#define LTC2992_USE_GPIO			1
```

Comment:
I still don't quite like this approach of guarding part functionalities using macro definitions. Moreover you enable all features by default here. I'd suggest dropping it completely, but let's see what others have to say.

Reply:
While waiting for other reviewers, I have followed your suggestion and dropped the guarding macro completely.